src/test/tools/pmemobjcli/pmemobjcli.c

```diff
@@ -391,6 +395,182 @@ parse_stage(void)
 	return stage;
 }
 
+/*
+ * pocli_reserv_id -- parse object's descriptor from action object
+ */
+static enum pocli_ret
+pocli_reserv_id(struct pocli_ctx *ctx, char *in, struct pobj_action **actp)
+{
+	char *input = strdup(in);
+	if (!input)
+		return POCLI_ERR_MALLOC;
+
+	if (!actp)
+		return POCLI_ERR_PARS;
+
+	struct pocli_args *args = pocli_args_alloc(NULL, input, ".");
+	if (!args)
+		return POCLI_ERR_PARS;
+	enum pocli_ret ret = POCLI_RET_OK;
+
+	if (strcmp(args->argv[0], "g") != 0) {
+		ret = POCLI_ERR_PARS;
+		goto out;
+	}
+	size_t size = pmemobj_root_size(ctx->pop);
+	for (int i = 1; i < args->argc; i++) {
+
+		unsigned ind;
+		char c;
+		int n = sscanf(args->argv[i], "%u%c", &ind, &c);
+		if (n != 1) {
+			ret = POCLI_ERR_PARS;
+			goto out;
+		}
+		size_t max_ind = size / sizeof(PMEMoid);
+		if (!max_ind || ind >= max_ind) {
+			ret = POCLI_ERR_PARS;
+			goto out;
+		}
+		*actp = &g_acts[ind];
+	}
+out:
+	free(input);
+	free(args);
+	return ret;
+}
+
+/*
+ * pocli_args_action -- parse action
+ */
+static enum pocli_ret
+pocli_args_act(struct pocli_ctx *ctx, struct pocli_args *args, int arg,
+	struct pobj_action **actp)
+{
+	assert(args != NULL);
+	assert(arg >= 0 && arg < args->argc);
+	assert(actp != NULL);
+	assert(ctx != NULL);
+
+	char *acts = args->argv[arg];
+
+	if (strcmp(acts, "0") == 0) {
+		*actp = NULL;
+	} else if (strcmp(acts, "NULL") == 0) {
+		*actp = NULL;
+	} else if (acts[0] == 'g') {
+		return pocli_reserv_id(ctx, args->argv[arg], actp);
+	} else {
+		return pocli_err(ctx, POCLI_ERR_PARS,
+			"invalid object specified -- '%s'\n", acts);
+	}
+	return POCLI_RET_OK;
+}
+
+/*
+ * pocli_args_action_publish -- parse action for publish
+ */
+static enum pocli_ret
+pocli_args_act_pub(struct pocli_ctx *ctx, struct pocli_args *args, int arg,
+	struct pobj_action **actp_pub)
+{
+	assert(args != NULL);
+	assert(arg >= 0 && arg < args->argc);
+	assert(actp_pub != NULL);
+	assert(ctx != NULL);
+
+	enum pocli_ret ret = POCLI_RET_OK;
+
+	char *acts_p = args->argv[arg];
+
+	if (strcmp(acts_p, "0") == 0) {
+		*actp_pub = NULL;
+	} else if (strcmp(acts_p, "NULL") == 0) {
+		*actp_pub = NULL;
+	} else if (acts_p[0] == 'g') {
+		char *input = strdup(acts_p);
+		if (!input)
+			return POCLI_ERR_MALLOC;
+
+		struct pocli_args *args = pocli_args_alloc(NULL, input, ".");
+		if (!args)
+			return POCLI_ERR_PARS;
+
+		size_t size = pmemobj_root_size(ctx->pop);
+		for (int i = 1; i < args->argc; i++) {
+
+			unsigned ind;
+			char c;
+			int n = sscanf(args->argv[i], "%u%c", &ind, &c);
+			if (n != 1) {
+				ret = POCLI_ERR_PARS;
+				goto out;
+			}
+			size_t max_ind = size / sizeof(PMEMoid);
+			if (!max_ind || ind >= max_ind) {
+				ret = POCLI_ERR_PARS;
+				goto out;
+			}
+			if (arg == 1) {
+				g_pub[0] = ind;
```

Comment:
this doesn't have to be a global variable, you can pass it as a function argument

Reply:
fixed